src/api/c/unary.cpp

```diff
@@ -673,6 +673,7 @@ static af_err af_check(af_array *out, const af_array in) {
         switch (type) {
```

Comment:
`type` is  f32 when the input is f16.

Reply:
need `        if (in_type == f16) { type = f16; }` before that line

Reply:
ahh good catch.